codex-rs/core/src/codex.rs

```diff
@@ -782,6 +782,36 @@ async fn submission_loop(
                     }
                 });
             }
+
+            Op::SummarizeContext => {
+                let sess = match sess.as_ref() {
+                    Some(sess) => sess,
+                    None => {
+                        send_no_session_event(sub.id).await;
+                        continue;
+                    }
+                };
+
+                // Create a summarization request as user input
+                const SUMMARIZATION_PROMPT: &str = concat!(
+                    "Please provide a summary of our conversation so far, highlighting key points, ",
```

Comment:
I would favor a raw string (`r#`) over `concat!()` here (unless you really want this to be one line of text?): https://doc.rust-lang.org/rust-by-example/std/str.html#literals-and-escapes